guides/documentation.md

```diff
@@ -61,3 +61,34 @@ Documentation should be linked to from an appropriate landing page
 (for example, [this repository's README](https://github.com/cfpb/development/blob/master/README.md))
 and, if possible, be discoverable through search
 (for example, [the cfgov-refresh docs search](https://cfpb.github.io/cfgov-refresh/search.html?q=testing)).
+
+The onboarding of new team members can be a great opportunity to review and
+update existing documentation to ensure that it meets these goals.
+
+## Recommendations for good documentation
+
+Source code repositories should contain a README that provides an overview of
+what they contain. See the
+[CFPB Open Source Project Template](https://github.com/cfpb/open-source-project-template)
+for a recommended example structure.
+
+Installation instructions should clearly present what a user might expect upon
+successful completion of a setup process.
+For example, include expected console output or screenshots of webpages or graphical interfaces.
+When setup instructions include the running of console commands,
+make sure to specify within which directory those commands should be run.
+
+Make sure that project documentation includes information on how to install any
+required third-party dependencies.
+Be as specific as possible about the required versions of any external dependencies.
+
+When referencing external links, try to make the references as specific as possible,
+for example by using page hash links
+([like this](https://github.com/cfpb/development/#guides)) to target specific places on the page.
+When referencing specific lines in source code,
+consider linking to specific releases or commit hashes instead of the master branch
```

Comment:
I don't know if [we agreed on this, did we?](https://github.com/cfpb/cfgov-refresh/pull/4727#issuecomment-452448726)

Reply:
I agree with Andy that it should be considered. I think what he wrote here is fine, but maybe add a sentence describing the other POV on it?

Reply:
How about

> When referencing specific lines in source code, consider linking to specific releases or commit hashes instead of the master branch ([this](https://github.com/cfpb/cfgov-refresh/blob/7.2.2/tox.ini#L105) or [this](https://github.com/cfpb/cfgov-refresh/blob/fb16e906bc4935669f880c270ccf4e32b930b068/tox.ini#L105) instead of [this](https://github.com/cfpb/cfgov-refresh/blob/master/tox.ini#L105)) when the file being linked to is likely to change often and a specific line reference would be hard to maintain. Links to the master branch of source code should be used as long as the documentation is updated appropriately when the source files are.

Reply:
Seems fine to me.